hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -126,18 +128,27 @@ public DeletedBlockLogImpl(ConfigurationSource conf,
   }
 
   @Override
-  public List<DeletedBlocksTransaction> getFailedTransactions()
+  public List<DeletedBlocksTransaction> getFailedTransactions(int count)
       throws IOException {
     lock.lock();
     try {
       final List<DeletedBlocksTransaction> failedTXs = Lists.newArrayList();
       try (TableIterator<Long,
           ? extends Table.KeyValue<Long, DeletedBlocksTransaction>> iter =
                deletedBlockLogStateManager.getReadOnlyIterator()) {
-        while (iter.hasNext()) {
-          DeletedBlocksTransaction delTX = iter.next().getValue();
-          if (delTX.getCount() == -1) {
-            failedTXs.add(delTX);
+        if (count <= 0) {
```

Comment:
We should use LIST_ALL_FAILED_TRANSACTIONS check here.

Reply:
Updated